projects/chaos/src/gpio_fsm.c

```diff
@@ -0,0 +1,101 @@
+#include "gpio_fsm.h"
+
+#include <stdbool.h>
+
+#include "chaos_config.h"
+#include "chaos_events.h"
+#include "event_queue.h"
+#include "fsm.h"
+#include "gpio.h"
+#include "gpio_seq.h"
+#include "misc.h"
+
+#define GPIO_FSM_SLEW_RATE 10
+
+static FSM s_gpio_fsm;
+
+FSM_DECLARE_STATE(idle);
```

Comment:
Should this follow the naming scheme? Maybe at least `state_[x]`.

Reply:
Done.